openlibrary/plugins/openlibrary/js/edit.js

```diff
@@ -70,30 +70,163 @@ export function initRoleValidation() {
     });
 }
 
-export function initIdentifierValidation() {
+/**
+ * isIsbnDedupe takes an isbn string and returns true if the given ISBN
+ * is already added to this edition.
+ * @params isbn - ISBN string duplication checking
+ * @returns true if the given ISBN is already added to the edition
+ */
+function isIsbnDupe(isbn) {
+    const isbnEntries = document.querySelectorAll('.isbn_10, .isbn_13');
+    for (let i = 0; i < isbnEntries.length; i++) {
+        if (isbn === isbnEntries[i]['value']) {
+            return true;
+        }
+    }
+    return false;
```

Comment:
You may want to use more modern JS syntax like the Array.some() function. 
It might look like this:
`return isbnEntries.some(entry => entry['value'] === isbn)`


https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

Reply:
I appreciate this and will update the code. Thanks for the input on how I can make it better. :)